ethereum-side/NFTLock.sol

```diff
@@ -12,173 +12,97 @@ contract NFTLock is INFTLock, IERC721Receiver {
     struct NFT {
         uint externalTokenId;
         address tokenContractAddress;
-        State state;
+        address depositor;
     }
 
-    mapping (address => mapping (uint256 => NFT)) tokens;
-    mapping (address => uint256) totalTokensByAddress;
-
-    uint256 totalDepositedTokens;
-
-    address private admin;
+    mapping (uint256 => NFT) lockedTokenById;
 
+    uint256[] releasedIds;
 
-    /** Stages that every NFT gets trough.
-      *   deposited - During this state only lock is allowed.
-      *   locked - During this stage tezos wNFT mint is allowed.
-      *   wrapMinted - At this stage Tezos wNFT was minted, the unlock is not allowed.
-      *   wrapBurned - At this stage Tezos wNFT was burned, the unlock is now allowed.
-      *   unlocked - This stage gives owner opportunity to withdraw its NFT.
-    */
-    enum State { deposited, locked, wrapMinted, wrapBurned, unlocked }
+    uint256 totalDepositedTokens;
```

Comment:
Seeing what this variable is for, it should rather be called `totalLockedTokens`